src/coverlet.collector/DataCollector/CoverletCoverageDataCollector.cs

```diff
@@ -0,0 +1,179 @@
+﻿namespace Coverlet.Collector.DataCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Xml;
+    using coverlet.collector.DataCollector;
+    using Coverlet.Collector.Utilities;
+    using Coverlet.Collector.Utilities.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+
+    /// <summary>
+    /// Coverlet coverage out-proc data collector.
+    /// </summary>
+    [DataCollectorTypeUri(CoverletConstants.DefaultUri)]
+    [DataCollectorFriendlyName(CoverletConstants.FriendlyName)]
+    public class CoverletCoverageCollector : DataCollector
+    {
+        private readonly TestPlatformEqtTrace eqtTrace;
+        private DataCollectionEvents events;
+        private TestPlatformLogger logger;
+        private XmlElement configurationElement;
+        private DataCollectionSink dataSink;
+        private DataCollectionContext dataCollectionContext;
+        private CoverageManager coverageManager;
+        private ICoverageWrapper coverageWrapper;
+
+        public CoverletCoverageCollector() : this(new TestPlatformEqtTrace(), new CoverageWrapper())
+        {
+        }
+
+        internal CoverletCoverageCollector(TestPlatformEqtTrace eqtTrace, ICoverageWrapper coverageWrapper) : base()
+        {
+            this.eqtTrace = eqtTrace;
+            this.coverageWrapper = coverageWrapper;
+        }
+
+        /// <summary>
+        /// Initializes data collector
+        /// </summary>
+        /// <param name="configurationElement">Configuration element</param>
+        /// <param name="events">Events to register on</param>
+        /// <param name="dataSink">Data sink to send attachments to test platform</param>
+        /// <param name="logger">Test platform logger</param>
+        /// <param name="environmentContext">Environment context</param>
+        public override void Initialize(
+            XmlElement configurationElement,
+            DataCollectionEvents events,
+            DataCollectionSink dataSink,
+            DataCollectionLogger logger,
+            DataCollectionEnvironmentContext environmentContext)
+        {
+            if (this.eqtTrace.IsInfoEnabled)
+            {
+                this.eqtTrace.Info("Initializing {0} with configuration: '{1}'", CoverletConstants.DataCollectorName, configurationElement?.OuterXml);
```

Comment:
[Question]Where we can see this logs?

Reply:
When you add switch `--diag:"C:/codebase/log.txt"` to `dotnet test`, all processes,` vstest.console`, `testhost` and `datacollector` have their own log files created. All these traces go in those files.

Reply:
Great thanks!

Reply:
So are related to verbosity level, right?

Reply:
Yes. These are related to verbosity levels